### PR TITLE
Validate all the schemas

### PR DESCRIPTION
**Pull request by coryodaniel**
Validates during commit:

* All generated JSON Schemas are valid (catches weird stuff put into massdriver.yaml)
* Validates your development params _match_ your schema-params ... if they dont, that means you can't _possibly_ have tested this locally.
* Validates your connection params _match_ your schema-connections ... if they dont, that means you can't _possibly_ have tested this locally.

### CODE REVIEW

**Review by jaketf — APPROVED** [review]
I love this :)

**Comment by coryodaniel** [comment]
@jaketf

File: pkg/bundle/templates/terraform/.pre-commit-config.yaml [hunk 1]
```diff
@@ -17,4 +17,36 @@ repos:
...
+        language: system
+        pass_filenames: false
+      - id: dev-params-validation
+        name: Validate development input params
+        entry: mass schema validate -s ./schema-params.json -d ./src/dev.params.tfvars.json
+        language: system
+        pass_filenames: false
+      - id: conn-params-validation
+        name: Validate development connection params
+        entry: mass schema validate -s ./schema-connections.json -d ./src/dev.connections.tfvars.json
```
Comment (WillBeebe) [inline pkg/bundle/templates/terraform/.pre-commit-config.yaml:50]: 💯

File: pkg/bundle/templates/terraform/.pre-commit-config.yaml [hunk 2]
```diff
@@ -17,4 +17,36 @@ repos:
...
+        language: system
+        pass_filenames: false
+      - id: valid-schema-artifacts
+        name: Validate schema artifacts
+        entry: mass schema validate -s https://json-schema.org/draft-07/schema -d ./schema-artifacts.json
+        language: system
+        pass_filenames: false
+      - id: dev-params-validation
+        name: Validate development input params
+        entry: mass schema validate -s ./schema-params.json -d ./src/dev.params.tfvars.json
```
Comment (WillBeebe) [inline pkg/bundle/templates/terraform/.pre-commit-config.yaml:45]: 💯

Comment (chrisghill) [reply pkg/bundle/templates/terraform/.pre-commit-config.yaml:45]: only thing that might get us here is if the `md_metadata` block is in the `dev.params.tfvars.json` block and it causes it to be invalid.

**Comment by coryodaniel** [comment]
Reordered this pc w/ `local` first, so we can _build_ the bundle, this lets us use `tf validate` later in the process.

File: pkg/bundle/templates/terraform/.pre-commit-config.yaml (hunk 2 above)
Comment (coryodaniel) [reply pkg/bundle/templates/terraform/.pre-commit-config.yaml:27]: Then that means that the developer didnt try running it.

**Comment by coryodaniel** [comment]
@jaketf @WillBeebe @chrisghill added a workflow that will validate the build on pr. I'm going to push this into all of our bundles use the bundle manager repo. Could use some eyes to see if anything is goofy. POC run [here](https://github.com/massdriver-cloud/aws-rds-mysql/runs/6853291902?check_suite_focus=true)

**Review by xpositivityx — APPROVED** [review]

**Pull request merged**